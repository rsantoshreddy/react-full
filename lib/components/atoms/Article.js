import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import storeProvider from "../providers/storeProvider";

const display = date => new Date(date).toDateString();

class Article extends PureComponent {
	render() {
		const { article, author } = this.props;
		const { title, date, body } = article;
		const { website, firstName, lastName } = author;
		return (
			<div>
				<h1>{title}</h1>
				<p>{display(date)}</p>
				<p>
					<a href={website}>
						{firstName} {lastName}
					</a>
				</p>
				<p>{body}</p>
			</div>
		);
	}
}
Article.propTypes = {
	article: PropTypes.shape({
		title: PropTypes.string.isRequired,
		body: PropTypes.string.isRequired,
		date: PropTypes.string.isRequired
	})
};
const mapStateToProps = (state, props) => {
	return {
		author: state.lookUpAuthor(props.article.authorId)
	};
};
export default storeProvider(mapStateToProps)(Article);
