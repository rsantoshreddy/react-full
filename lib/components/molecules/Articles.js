import React, { PureComponent } from "react";
import Article from "../atoms/Article";

class Articles extends PureComponent {
	render() {
		const { articles, store } = this.props;
		const articlesArray = Object.keys(articles).map(key => articles[key]);
		return (
			<div>
				{articlesArray.map((article, key) => (
					<Article article={article} key={key} />
				))}
			</div>
		);
	}
}

export default Articles;
