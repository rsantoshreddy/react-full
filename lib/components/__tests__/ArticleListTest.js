import React from "react";
import Articles from "../molecules/Articles";
// import ReactTestRenderer from "react-test-renderer";
import { shallow } from "enzyme";
import Article from "../atoms/Article";

describe("Articles", () => {
	const data = {
		articles: {
			a: { id: "a" },
			b: { id: "b" }
		}
		// store: {
		// 	lookUpAuthor: jest.fn(() => ({}))
		// }
	};

	//this is to fix the error when Article component is rendered with Articles component with jest
	//Article.propTypes = {};

	it("renders correctly", () => {
		// const tree = ReactTestRenderer.create(<Articles {...data} />).toJSON();
		// expect(tree.children.length).toEqual(2);
		// expect(tree).toMatchSnapshot();
		const wrapper = shallow(<Articles {...data} />);
		expect(wrapper.find("ArticleContainer").length).toBe(2);
		expect(wrapper).toMatchSnapshot();
	});
});
