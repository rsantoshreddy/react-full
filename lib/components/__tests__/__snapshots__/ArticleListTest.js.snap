// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <ArticleContainer
        article={
            Object {
                "id": "a",
              }
        }
    />
    <ArticleContainer
        article={
            Object {
                "id": "b",
              }
        }
    />
</div>,
  "nodes": Array [
    <div>
      <ArticleContainer
            article={
                  Object {
                        "id": "a",
                      }
            }
      />
      <ArticleContainer
            article={
                  Object {
                        "id": "b",
                      }
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Articles
      articles={
            Object {
                  "a": Object {
                    "id": "a",
                  },
                  "b": Object {
                    "id": "b",
                  },
                }
      }
/>,
    "_forcedUpdate": false,
    "_instance": Articles {
      "context": Object {},
      "props": Object {
        "articles": Object {
          "a": Object {
            "id": "a",
          },
          "b": Object {
            "id": "b",
          },
        },
      },
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div>
      <ArticleContainer
            article={
                  Object {
                        "id": "a",
                      }
            }
      />
      <ArticleContainer
            article={
                  Object {
                        "id": "b",
                      }
            }
      />
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Articles
    articles={
        Object {
            "a": Object {
              "id": "a",
            },
            "b": Object {
              "id": "b",
            },
          }
    }
/>,
}
`;
